core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -562,4 +581,53 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
             // Netty will close the connection automatically on an IOException.
         }
     }
+
+    private void applyOutlierDetection(ClientRequestContext ctx) {
+        if (outlierDetectingRule == null || !isAcquirable()) {
+            return;
+        }
+        assert outlierDetector != null;
+        if (outlierDetector.isOutlier()) {
+            return;
+        }
+
+        ctx.log().whenComplete().thenAccept(this::detectOutlier);
+    }
+
+    private void detectOutlier(RequestLog log) {
+        final RequestContext context = log.context();
+        if (!context.eventLoop().inEventLoop()) {
+            // Execute in the event loop to prevent a connection from being marked as an outlier as a request is
+            // about to be sent.
+            detectOutlier(log);
```

Comment:
I'm guessing you intended an early return here

```suggestion
            detectOutlier(log);
            return;
```

Reply:
Oops... 🤦‍♂️